packages/quantic/setup-quantic.js

```diff
@@ -0,0 +1,108 @@
+const fs = require('fs');
+const pathlib = require('path');
+
+function copyFileSync(source, target) {
+  try {
+    let targetFile = target;
+    if (fs.existsSync(target) && fs.lstatSync(target).isDirectory()) {
+      targetFile = pathlib.join(target, pathlib.basename(source));
+    }
+
+    fs.writeFileSync(targetFile, fs.readFileSync(source));
+  } catch (err) {
+    console.error(`Failed to copy file at the following path ${source}`);
+    process.exit(1);
+  }
+}
+
+function copyFolderRecursiveSync(source, target) {
+  try {
+    const targetFolder = pathlib.join(target, pathlib.basename(source));
+    if (!fs.existsSync(targetFolder)) {
+      fs.mkdirSync(targetFolder);
+    }
+
+    if (fs.lstatSync(source).isDirectory()) {
+      const files = fs.readdirSync(source);
+      files.forEach(function (file) {
+        let sourceFilePath = pathlib.join(source, file);
+        if (fs.lstatSync(sourceFilePath).isDirectory()) {
+          copyFolderRecursiveSync(sourceFilePath, targetFolder);
+        } else {
+          copyFileSync(sourceFilePath, targetFolder);
+        }
+      });
+    }
+  } catch (err) {
+    console.error(`Failed to copy folder at the following path ${source}`);
+    process.exit(1);
+  }
+}
+
+function createQuanticDirectory(path) {
+  try {
+    const quanticDirectoryPath = pathlib.join(path, 'quantic');
+    if (fs.existsSync(quanticDirectoryPath)) {
+      fs.rmdirSync(quanticDirectoryPath, {recursive: true});
+    }
```

Comment:
Should this really delete the repository that already exists?
I don't exactly know what should be done instead but let's assume there was a fuckup or something with the folder this script found/chose, deleting it might cause issues.

Is it possible perhaps to just copy/overwrite what's in the folder instead?

I'd be okay overwriting files we're supposed to control, but let's say someone has made custom quantic components and added them to this folder, this would delete them.

Reply:
Good point👌 I updated this to avoid deleting anything and to just overwrite the files we're supposed to control.